src/metabase/util.clj

```diff
@@ -870,12 +870,26 @@
                          (Math/log 10))))))
 
 (defn update-when
-  "Like clojure.core/update but does not create a new key if it does not exist."
+  "Like clojure.core/update but does not create a new key if it does not exist.
+   Useful when you don't want to create cruft."
   [m k f & args]
   (if (contains? m k)
     (apply update m k f args)
     m))
 
+(defn update-in-when
+  "Like clojure.core/update-in but does not create new keys if they do not exist.
+   Useful when you don't want to create cruft."
+  [m k f & args]
+  (if (not= ::not-found (reduce (fn [m k]
```

Comment:
What's the difference between this and `(not= ::not-found (get-in m k ::not-found))`